pandas/tests/indexes/multi/test_setops.py

```diff
@@ -541,6 +541,19 @@ def test_union_keep_ea_dtype(any_numeric_ea_dtype, val):
     tm.assert_index_equal(result, expected)
 
 
+def test_union_keep_ea_dtype_duplicates_right(any_numeric_ea_dtype):
+    # GH#48591
+    arr1 = Series([4, 2], dtype=any_numeric_ea_dtype)
+    arr2 = Series([2, 1, 1], dtype=any_numeric_ea_dtype)
```

Comment:
Could you test with `NA` in this array too?

Reply:
Will add, but depends on #46608 and #48626, this fails currently because the sorting does not work right now